ml-agents/mlagents/trainers/tests/mock_brain.py

```diff
@@ -45,11 +45,11 @@ def create_mock_steps(
     reward = np.array(num_agents * [1.0], dtype=np.float32)
     interrupted = np.array(num_agents * [False], dtype=np.bool)
     agent_id = np.arange(num_agents, dtype=np.int32)
-    behavior_spec = BehaviorSpec(
-        observation_shapes,
-        ActionType.DISCRETE if discrete else ActionType.CONTINUOUS,
-        action_shape,
-    )
+    if discrete:
```

Comment:
I haven't looked at this code in ages so I don't remember exactly how it's used. But seems like we unpack the action spec in create_steps_from_behavior_spec and then recreate it here - might be easier to just make this method take an ActionSpec instead. (but maybe not if it's used in a lot of places).

Reply:
I think the main downside is that it would make some tests a bit harder to parametrize (since it's not just `discrete = True,False` but you'd have to create an ActionSpec in the test and pass it).

But actually it seems like the `discrete=` parameter is actually never used in the tests, so maybe just having a default `ActionSpec` would be pretty much the same.

Reply:
Updated make_fake_trajectory and make_fake_step to take ActionSpec